api/src/pcapi/scripts/format_user_phone_numbers.py

```diff
@@ -0,0 +1,147 @@
+import argparse
+from datetime import datetime
+import logging
+import os
+import warnings
+
+from sqlalchemy.orm import load_only
+
+
+warnings.filterwarnings("ignore")
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def get_users(from_, to):  # type: ignore [no-untyped-def]
+    users = (User.query.options(load_only("id", "phoneNumber")).order_by(User.id))[from_:to]
+    return users
+
+
+def process_users_batch(dry, users):  # type: ignore [no-untyped-def]
+    changed = 0
+    errors = 0
+    none = 0
+    ok = 0
+
+    for user in users:
+        if user.phoneNumber is None:
```

Comment:
Il y a aussi des chaînes vides dans les numéros de téléphone.
Soit tu les prends en compte ici (`if not user.phoneNumber:`), soit par souci d'homogénéité tu les remets à `None`.

Sur la production aujourd'hui : 
```
pcapi-production> select "phoneNumber", count(*) from "user" where "phoneNumber" is null or "phoneNumber" = '' group by "phoneNumber";
+-------------+---------+
| phoneNumber | count   |
|-------------+---------|
|             | 476     |
| <null>      | 1942967 |
+-------------+---------+
```

Mais a priori il y a encore du code qui permet d'avoir une chaîne vide, puisque des comptes créés en juillet sont concernés : 
```
pcapi-production> select id, "phoneNumber", "dateCreated" from "user" where "phoneNumber" = '' order by id desc limit 5;
+---------+-------------+----------------------------+
| id      | phoneNumber | dateCreated                |
|---------+-------------+----------------------------|
| 3343426 |             | 2022-07-16 15:08:06.838843 |
| 3320169 |             | 2022-07-09 19:43:06.06763  |
| 3317668 |             | 2022-07-09 09:37:02.830332 |
| 3307515 |             | 2022-07-06 22:57:39.820295 |
| 3304837 |             | 2022-07-06 15:18:50.485807 |
+---------+-------------+----------------------------+
```

Reply:
le code de la PR devrait interdire toute nouvelle chaine vide dans le champs `User.phoneNumber`.

j'adapterai le script de rattrapage pour les chaines vides en compte.